exp/api/v1alpha3/azuremanagedcontrolplane_types.go

```diff
@@ -94,6 +98,23 @@ type AADProfile struct {
 	AdminGroupObjectIDs []string `json:"adminGroupObjectIDs"`
 }
 
+// APIServerAccessProfile - access profile for AKS API server.
+type APIServerAccessProfile struct {
+	// AuthorizedIPRanges - Authorized IP Ranges to kubernetes API server.
+	// +optional
+	AuthorizedIPRanges []string `json:"authorizedIPRanges,omitempty"`
+	// EnablePrivateCluster - Whether to create the cluster as a private cluster or not.
+	// +optional
+	EnablePrivateCluster bool `json:"enablePrivateCluster,omitempty"`
```

Comment:
If this field is optional, I believe a bool pointer `*bool` would be better. Same for `EnablePrivateClusterPublicFQDN`. WDYT?

Reply:
@nprokopic Thanks for your feedback! I'm pretty new to golang and cluster api, and my (limited) understanding of using a pointer is that we could differentiate between a zero value and a not specified value, but then we have to deal with `nil`. For `EnablePrivateCluster`, I think explicitly passing in a zero value (`false`) is equivalent to not passing it (both tells Azure to not enable private cluster). Therefore I'm not aware of the benefit of using a pointer.

Could you help explain a bit more why a bool pointer is better? I think I must have missed something and I'd love to learn :) Thank you!

Reply:
It is part of the k8s API conventions:

> Therefore, we ask that pointers always be used with optional fields that do not have a built-in nil value.

https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#optional-vs-required

Reply:
Got it, thanks for the reference! I'll change the fields to use pointers.

Reply:
> It is part of the k8s API conventions

I vaguely remembered something like that, and that we had a similar discussion for some other CAPZ type some time ago, just couldn't find it.

Thanks Cecile for referencing this.